ttnn/cpp/ttnn/operations/conv/conv2d/conv2d.cpp

```diff
@@ -790,7 +799,7 @@ static void adjust_conv_op_config_for_auto_shard(
         shard_orientation,
         compute_grid_size);
 
-    if (conv_config.act_block_h_override == 0 && conv_config.shard_layout != TensorMemoryLayout::WIDTH_SHARDED) {
+    if (conv_config.act_block_div == 1 && conv_config.shard_layout != TensorMemoryLayout::WIDTH_SHARDED) {
         if (in_channels <= constants::TILE_WIDTH / 2 && conv_config.input_channels_alignment == constants::TILE_WIDTH &&
             !is_mm_conv && conv_config.shard_layout == TensorMemoryLayout::HEIGHT_SHARDED && input_tensor_layout == Layout::ROW_MAJOR) {
```

Comment:
check on line 802 for width sharding should now be removed as act_block_div now affects all shardings.
Also this should check if act_block_div is nullptr (which should be the default), and than tweak it only if user didn't set it (if it was nullptr)